test/sqlite/index.js

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const sqlite = require("sqlite3");
+
+async function main() {
+  const db = new sqlite.Database(":memory:");
+
+  const aCallback = (err) => console.log("Err", err);
+
+  // We are serializing the commands just to get the same [id, parent_id] pairs
+  // for the events in the appmap snapshot in every test run.
+  db.serialize(function () {
+    // Database
+    // run
+    db.run("SELECT 'Database.run'");
+    db.run("SELECT 'Database.run with callback'", aCallback);
+    db.run("SELECT 'Database.run with params', ?, ?", 123, "abc");
+    db.run("SELECT 'Database.run with params and callback', ?, ?", 123, "abc", aCallback);
+    db.run("SELECT 'Database.run with object param', $foo, $bar", { $foo: 123, $bar: "abc" });
+    // all
+    db.all("SELECT 'Database.all'");
+    db.all("SELECT 'Database.all with callback'", aCallback);
+    // get
+    db.get("SELECT 'Database.get'");
+    db.get("SELECT 'Database.get with callback'", aCallback);
+    // each
+    db.each("SELECT 'Database.each with row callback'", aCallback);
+    db.each(
+      "SELECT 'Database.each with row callback and completion callback'",
+      aCallback,
+      aCallback,
+    );
+    // exec
+    db.exec("SELECT 'Database.exec'");
+    db.exec("SELECT 'Database.exec with callback'", aCallback);
+
+    // Statement
+    // run
+    db.prepare("SELECT 'Stetement.run'").run().finalize();
+    db.prepare("SELECT 'Statement.run with callback'").run(aCallback).finalize();
+    db.prepare("SELECT 'Statement.run with params', ?, ?", 122, "abc").run().finalize();
+    db.prepare("SELECT 'Statement.run with params and callback', ?, ?", 123, "abc")
+      .run(aCallback)
+      .finalize();
+    db.prepare("SELECT 'Statement.run with object param', $foo, $bar", {
+      $foo: 123,
+      $bar: "abc",
+    })
+      .run()
+      .finalize();
+    // all
+    db.prepare("SELECT 'Statement.all'").all().finalize();
+    db.prepare("SELECT 'Statement.all with callback'").all(aCallback).finalize();
+    // get
+    db.prepare("SELECT 'Statement.get'").get().finalize();
+    db.prepare("SELECT 'Statement.get with callback'").get(aCallback).finalize();
+    // each
+    db.prepare("SELECT 'Statement.each with row callback'", aCallback).each().finalize();
+    db.prepare("SELECT 'Statement.each with row callback and completion callback'")
+      .each(aCallback, aCallback)
+      .finalize();
```

Comment:
You should also make sure if you run the same prepared statement twice it appears twice in the appmap.